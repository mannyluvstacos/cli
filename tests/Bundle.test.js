const { execFile } = require('child_process');

describe("Bundle", () => {
  const twoCommands = `${__dirname}/context/twoCommands.js`;
  const fullCommands = `${__dirname}/context/fullCommands.js`;

  it("should display help by default", done => {
    execFile("node", [twoCommands], (error, stdout) => {
      expect(stdout).toMatchSnapshot();
      done();
    });
  });

  it("should display help", done => {
    execFile("node", [twoCommands, "--help"], (error, stdout) => {
      expect(stdout).toMatchSnapshot();
      done();
    });
  });

  it("should display command help", done => {
    execFile("node", [twoCommands, "command1", "--help"], (error, stdout) => {
      expect(stdout).toMatchSnapshot();
      done();
    });
  });

  it("should execute command", done => {
    execFile("node", [twoCommands, "command1"], (error, stdout) => {
      expect(stdout).toEqual("command1 output");
      done();
    });
  });

  it("should display descriptions", done => {
    execFile("node", [fullCommands, "command3", "--help"], (error, stdout) => {
      expect(stdout).toMatchSnapshot();
      done();
    });
  });
});
