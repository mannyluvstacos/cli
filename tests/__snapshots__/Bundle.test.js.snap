// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bundle should display command help 1`] = `
"Usage: command1 [options]

Options:
  -h, --help  output usage information
command1 help"
`;

exports[`Bundle should display descriptions 1`] = `
"Usage: command3 [options] <arg1> <arg2> [arg3] [arg4]

command3 description

Arguments:

  arg1            First argument
  arg2            
  arg3            Third argument
  arg4            

Options:
  --opt1 <value>  Option 1 (default: \\"foo\\")
  --opt2          Option 2
  --opt3          
  -h, --help      output usage information
command3 help"
`;

exports[`Bundle should display help 1`] = `
"Usage: twoCommands [options] [command]

Options:
  -h, --help  output usage information

Commands:
  command1
  command2
"
`;

exports[`Bundle should display help by default 1`] = `
"Usage: twoCommands [options] [command]

Options:
  -h, --help  output usage information

Commands:
  command1
  command2
"
`;
